packages/iTwinUI-react/src/core/AvatarGroup/AvatarGroup.tsx

```diff
@@ -24,55 +24,61 @@ export type UserIconGroupProps = {
    */
   animated?: boolean;
   /**
-   * Size of user and count icons.
+   * Size of avatars and count avatars.
    * @default 'small'
    */
-  iconSize?: 'small' | 'medium' | 'large' | 'x-large';
+  avatarSize?: 'small' | 'medium' | 'large' | 'x-large';
   /**
-   * User Icons in the group.
+   * Avatars in the group.
    */
   children: React.ReactNode;
   /**
-   * User Count Icon props.
+   * Count Avatar props.
    */
-  countIconProps?: React.ComponentPropsWithRef<'div'>;
+  countAvatarProps?: React.ComponentPropsWithRef<'div'>;
 } & Omit<CommonProps, 'title'>;
+
+/**
+ * @deprecated Since v2, this has been renamed to `AvatarGroupProps` (Use with `AvatarGroup`)
+ */
+export type UserIconGroupProps = AvatarGroupProps;
+
 /**
- * Group User Icons together.
+ * Group Avatars together.
  *
- * User Icons stacking is based on `maxIcons` count. If you provide 8 User Icons and keep default 5 `maxIcons` count,
- * this component will show 5 User Icons and User Count Icon with "3" in it.
+ * Avatars stacking is based on `maxAvatars` count. If you provide 8 Avatars and keep default 5 `maxAvatars` count,
+ * this component will show 5 Avatars and Count Avatar with "3" in it.
  *
- * You can add custom User Count Icon behavior by using `countIconProps`.
+ * You can add custom Count Avatar behavior by using `countAvatarProps`.
  *
  * @example
- * <UserIconGroup iconSize='medium'>
- *  <UserIcon
+ * <AvatarGroup avatarSize='medium'>
+ *  <Avatar
  *    abbreviation="TR"
  *    backgroundColor={getUserColor("Terry Rivers")}
  *    title="Terry Rivers"
  *  />
- *  <UserIcon
+ *  <Avatar
  *    abbreviation="RM"
  *    backgroundColor={getUserColor("Robin Mercer")}
  *    title="Robin Mercer"
  *  />
- *  <UserIcon
+ *  <Avatar
  *    abbreviation="JM"
  *    backgroundColor={getUserColor("Jean Mullins")}
  *    title="Jean Mullins"
  *  />
- * </UserIconGroup>
+ * </AvatarGroup>
  */
-export const UserIconGroup = (props: UserIconGroupProps) => {
+export const AvatarGroup = (props: AvatarGroupProps) => {
   const maxLength = 99;
   const {
     children,
     animated = false,
     stacked = true,
-    maxIcons = 5,
-    iconSize = 'small',
-    countIconProps,
+    maxAvatars = 5,
+    avatarSize = 'small',
+    countAvatarProps,
```

Comment:
Why does `AvatarGroup` need to be backwards compatible with `UserIcon` when we still have `UserIconGroup`?

Reply:
> Why does `AvatarGroup` need to be backwards compatible with `UserIcon` when we still have `UserIconGroup`?

I think Mayank explained it here: https://github.com/iTwin/iTwinUI-react/pull/902#discussion_r1010713129. I guess we want to have as less breaking changes as possible to encourage users to upgrade to v2. So all users have to do is rename `UserIconGroup` -> `AvatarGroup` and `UserIcon` -> `Avatar` and everything should work.

But I think, we would want to make this breaking change of new prop names at some point. And according to [semver](https://semver.org/#summary), `MAJOR` version is for breaking changes like this and `MINOR` for backward compatible changes.

![image](https://user-images.githubusercontent.com/45748283/199319957-65ec3d71-2308-4355-83b1-28cfd295cf33.png)

So we should include these changes in v2? Or maybe I misunderstood how semver works?

Reply:
> These prop names need to remain the same as before to keep backwards compatibility with `UserIcon`.

I reverted those changes for now, like you suggested. I can re-add it if we decide to do that.

Reply:
When I opened the original issue, I laid out the plan as: deprecate in v2, remove in v3.

A deprecation doesn't need to happen in a major version, it's just convenient to do it now because the components were renamed in css v1.